src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricManagedIdentitySource.cs

```diff
@@ -16,34 +16,79 @@ internal class ServiceFabricManagedIdentitySource : AbstractManagedIdentity
         private const string ServiceFabricMsiApiVersion = "2019-07-01-preview";
         private readonly Uri _endpoint;
         private readonly string _identityHeaderValue;
+        private readonly bool _isFederated;
+        private static string _mitsEndpointFmiPath => "/metadata/identity/oauth2/fmi/credential";
 
-        internal static Lazy<HttpClient> _httpClientLazy;
-
-        public static AbstractManagedIdentity Create(RequestContext requestContext)
+        public static AbstractManagedIdentity Create(RequestContext requestContext, bool isFmiServiceFabric = false)
```

Comment:
I don't think this is right. The source detection should not interfere with `isFmi` flag. **The MSI source should be expected to be cached (statically, in memory) by MSAL.** 

In other words, each MSI request can either go to the normal endpoint or to the fmi endpoint.

Reply:
I dont think this is the case. The source detection is effectively turned off if `IsFmi` is enabled and only the FMI source will be used. 
See [ManagedIdentityClient](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/62095ff69c61eec22ea96a1efd9dd5327dd49e9d/src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs#L40)

This parameter specifically is to change this source to only do FMI.

Also, when you say cached (statically, in memory), can more that one of these source objects be cached? meaning, can there be one for regular SF and one for FMI?